pkg/gadgets/traceloop/tracer/perf_buffer.go

```diff
@@ -0,0 +1,250 @@
+//go:build linux
+// +build linux
+
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"fmt"
+	"reflect"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/cilium/ebpf/perf"
+	log "github.com/sirupsen/logrus"
+
+	"golang.org/x/sys/unix"
+)
+
+type perfEventHeader struct {
+	Type uint32
+	Misc uint16
+	Size uint16
+}
+
+type perfEventSample struct {
+	Size uint32
+}
+
+func getRingsNumber(perfReader *perf.Reader) int {
+	v := reflect.ValueOf(perfReader)
+	rings := reflect.Indirect(v).FieldByName("rings")
+
+	return rings.Len()
+}
+
+func readOverWritable(reader *reader, callback func(record perf.Record, size uint32) error) error {
+	// Get private field with reflect.
+	v := reflect.ValueOf(reader.reader)
+	rings := reflect.Indirect(v).FieldByName("rings")
+	ringsNumber := rings.Len()
+
```

Comment:
Can use `getRingsNumber()` here as well.

This `readOverwritable` looks like semi-hot code. Would it be an option (and make sense) to get the fields you access through reflection here exported in the original lib? If not, an optimization could be to take the offsets and access the fields directly afterwards - similarly to what I've done in #1081 recently. Unsure if it's necessary, though.